lib/helper/User.js

```diff
@@ -146,23 +122,12 @@ class User {
       '0x'
     );
 
-    return this.auxiliaryWeb3.eth.abi.encodeFunctionCall(
-      {
-        name: 'createAndAddModules',
-        type: 'function',
-        inputs: [
-          {
-            type: 'address',
-            name: 'proxyFactory'
-          },
-          {
-            type: 'bytes',
-            name: 'data'
-          }
-        ]
-      },
-      [this.proxyFactory, reducedModulesCreationData]
-    );
+    const createAndAddModules = new this.auxiliaryWeb3.eth.Contract(this.abiBinProvider.getABI('CreateAndAddModules'));
```

Comment:
Regarding L100, instead of hardcoded json, should it be available in an ABI?

Reply:
Yes, it seems I overlooked it as I only grepped for `encodeFunctionCall`. But what *is* it? It uses a contract constructor. Where can I get the required ABI from? 😕

Reply:
Maybe some contract from safe contracts. Not sure though. 🤔

Reply:
@pgev can you please help? Where can I get the ABI from without hard-coding it?

Reply:
In line 100 that is ...

Reply:
According to @pgev the contract is not available and the ABI has to be hard-coded. I added a comment.